lib/csetperm.gi

```diff
@@ -273,17 +282,34 @@ InstallGlobalFunction( AddCosetInfoStabChain, function( G, U, maxmoved )
 
 		      # do all points
 		      if t1<t-1 then
-			vert := G.orbit{ [ 1 .. t - 1 ] };
-			img := G.orbit[ t ];
-			while img <> G.orbit[ 1 ]  do
-			    vert := OnTuples( vert, G.transversal[ img ] );
-			    img  := img           ^ G.transversal[ img ];
-			od;
-			if ForAll( [ t1+1 .. t - 1 ], i -> not IsBound
-			  ( U.translabels[ pimg[ vert[ i ] ] ] ) )  then
+                        img := G.orbit[ t ];
+                        if t<=10*t1lim then
+                          vert := G.orbit{ [ 1 .. t - 1 ] };
+                          while img <> G.orbit[ 1 ]  do
+                              vert := OnTuples( vert, G.transversal[ img ] );
+                              img  := img           ^ G.transversal[ img ];
+                          od;
+                          found:=ForAll( [ t1+1 .. t - 1 ], i -> not IsBound
+                            ( U.translabels[ pimg[ vert[ i ] ] ] ) );
+                        else
+                          # avoid calculating tons of images of a long list
+                          # instead calculate images on the fly
+                          # this implicitly assumes that, if we get to so
+                          # long a list, failure will happen quickly.
+                          tl:=[];
+                          while img <> G.orbit[ 1 ]  do
+                              Add(tl,G.transversal[img]);
+                              img  := img           ^ G.transversal[ img ];
+                          od;
+                          found:=ForAll( [ t1+1 .. t - 1 ], i -> not IsBound
+                            ( U.translabels[ pimg[ vimg(G.orbit[ i ],tl) ] ] ) );
+                        fi;
+
+                        if found then
 			    U.repsStab[ t ][ s ] := true;
 			    index := index + lenflock;
 			fi;
+
 		      else
                         U.repsStab[ t ][ s ] := true;
                         index := index + lenflock;
```

Comment:
I was wondering why the indentation looked so weird here. And the answer is not that you're doing anything weird, it's that some of the _unchanged_ lines use tabs, and GitHub renders tabs with a different width than does your text editor (or my text editor, for that matter).

So there's no problem here 🙂

Reply:
And in fact it looks fine on the main pull request page; it's in the "[Files changed](https://github.com/gap-system/gap/pull/4574/files)" tab where it renders weirdly.